From e170f703a92d355e5d0d571e999cbff05d7b67a0 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Sun, 28 Mar 2010 12:27:41 +0300
Subject: [PATCH] Print full patch output when debhelper(1) DH_VERBOSE is on.
Organization: Private
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Add DH_VERBOSE environment variable check to functions
V3/quilt.pm::apply_quilt_patch() and Patch.pm::apply().

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 scripts/Dpkg/Source/Package/V3/quilt.pm |   15 ++++++++++++---
 scripts/Dpkg/Source/Patch.pm            |    6 ++++++
 2 files changed, 18 insertions(+), 3 deletions(-)

diff --git a/scripts/Dpkg/Source/Package/V3/quilt.pm b/scripts/Dpkg/Source/Package/V3/quilt.pm
index 58c494b..2dc1dd3 100644
--- a/scripts/Dpkg/Source/Package/V3/quilt.pm
+++ b/scripts/Dpkg/Source/Package/V3/quilt.pm
@@ -167,11 +167,20 @@ sub apply_quilt_patch {
     my $obj = Dpkg::Source::Patch->new(filename => $path);
 
     info(_g("applying %s"), $patch) if $opts{"verbose"};
+
+
+    my @options = [ '-s', '-t', '-F', '0', '-N', '-p1', '-u',
+		   '-V', 'never', '-g0', '-E', '-b',
+		   '-B', ".pc/$patch/" ];
+
+    #  Display all messages if debhelper(1) option DH_VERBOSE is on.
+    if ( defined $ENV{DH_VERBOSE}  and  $ENV{DH_VERBOSE}  > 0 ) {
+	$opts{options} = [ grep !/^(-s|--silent)$/, @{$opts{options}} ];
+    }
+
     $obj->apply($dir, timestamp => $opts{"timestamp"},
                 force_timestamp => 1, create_dirs => 1, remove_backup => 0,
-                options => [ '-s', '-t', '-F', '0', '-N', '-p1', '-u',
-                             '-V', 'never', '-g0', '-E', '-b',
-                             '-B', ".pc/$patch/" ]);
+                options => \@options);
 }
 
 sub get_patches {
diff --git a/scripts/Dpkg/Source/Patch.pm b/scripts/Dpkg/Source/Patch.pm
index 10f751c..37cace1 100644
--- a/scripts/Dpkg/Source/Patch.pm
+++ b/scripts/Dpkg/Source/Patch.pm
@@ -445,6 +445,12 @@ sub apply {
     $opts{"create_dirs"} = 1 unless exists $opts{"create_dirs"};
     $opts{"options"} ||= [ '-s', '-t', '-F', '0', '-N', '-p1', '-u',
             '-V', 'never', '-g0', '-b', '-z', '.dpkg-orig'];
+
+    #  Display all messages if debhelper(1) option DH_VERBOSE is on.
+    if ( defined $ENV{DH_VERBOSE}  and  $ENV{DH_VERBOSE}  > 0 ) {
+	$opts{options} = [ grep !/^(-s|--silent)$/, @{$opts{options}} ];
+    }
+
     $opts{"add_options"} ||= [];
     push @{$opts{"options"}}, @{$opts{"add_options"}};
     # Check the diff and create missing directories
-- 
1.7.0

